ITZSHOAIB/chukti
# Contract Deployment Error Handling

# Description
This pull request introduces a comprehensive error-handling mechanism for contract deployment to improve the stability and robustness of our smart contract deployments. The new changes include:

```deployError.ts```: This file adds a utility to handle specific deployment errors, ensuring we capture common issues like contract deployment failures, network errors, or invalid configurations.
* Enhanced error messages for better debugging and resolution guidance.
* Ensured that all error cases log useful information, providing clarity during troubleshooting.

## Related Issues
#62 


## Type of changes

- Added deployError.ts to centralize error handling for contract deployments.
- Modified existing deployment logic to integrate with deployError.ts.
- Ensured that errors are logged with detailed information for better debugging.

## Checklist:



- [x] I have checked that there isn't already a PR that solves the problem the same way.
- [x] I have read the [**CONTRIBUTING**](https://github.com/ITZSHOAIB/chukti/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have included a **changeset** according to the type of my changes

## Steps to Test:
1. Run the deployment script with incorrect contract data or a simulated failure scenario.
2. Verify that the new error messages provide clear guidance on the failure reason.
3. Ensure that no unhandled errors occur during the contract deployment process.
